lms/services/blackboard_api/service.py

```diff
@@ -53,12 +53,38 @@ def list_files(self, course_id):
         :raise ProxyAPIAccessTokenError: if we don't have a Blackboard API
             access token for the current user
         """
-        return self._http_service.get(
+        files = self._http_service.get(
             self._api_url(f"courses/uuid:{course_id}/resources"),
             oauth=True,
             schema=BlackboardListFilesSchema,
         ).validated_data
 
+        return [
+            {
+                "id": self._blackboard_document_url(file_["id"]),
+                "display_name": file_["name"],
+                "updated_at": file_["modified"],
+            }
+            for file_ in files
+        ]
```

Comment:
The `list_files()` method is now doing the work of renaming the `name` and `modified` fields and of excluding the `downloadUrl` field from the `list_files()` response (where that field isn't needed). This is because the `public_url()` method below now uses the same `BlackboardListFilesSchema` and `public_url()` _doesn't_ want this renaming and _does_ want `downloadUrl`.

This does unfortunately mean that knowledge of the Blackboard API's list files response format is now in both the schema and the service (both the `list_files()` and `public_url()` methods) and in the tests for both :man_shrugging: